core/src/main/java/org/apache/iceberg/AllDataFilesTable.java

```diff
@@ -53,47 +48,25 @@ MetadataTableType metadataTableType() {
     return MetadataTableType.ALL_DATA_FILES;
   }
 
-  public static class AllDataFilesTableScan extends BaseFilesTableScan {
+  public static class AllDataFilesTableScan extends BaseAllFilesTableScan {
 
-    AllDataFilesTableScan(TableOperations ops, Table table, Schema fileSchema) {
-      super(ops, table, fileSchema, MetadataTableType.ALL_DATA_FILES);
+    AllDataFilesTableScan(TableOperations ops, Table table, Schema schema) {
```

Comment:
I took a closer look and it seems it is not really `fileSchema`. It is more the metadata table schema before column pruning. That's the purpose of that variable in `BaseTableScan`.

Reply:
I think it was called `fileSchema` because it was the projection of the `DataFile` struct.

Reply:
Okay, that's why. I think it still makes sense to rename as I got rid of a separate `fileSchema` var.